### DISCUSSION TITLE
Handling long URLs for batches

### DISCUSSION THREAD

**Discussion by kimroen** [trpc/trpc#1203] [Q&A] [answered]
I had a case recently where I was (accidentally) causing a very long batch URL to be generated, which was rejected by the server (431 Request Header Fields Too Large). The length of this request URL was 246490 characters (2629 individual procedures - superjson adds quite a bit to the query), and as far as I can see a common limit for URLs is 2048 characters.

It seems to me that even in non-extreme cases such as the one I'm describing here, you could easily hit this URL limit with batching and superjson. Are there any known approaches we might look at for handling this? Perhaps automatically sending these as POST-requests, or automatically splitting into appropriately sized GET-requests?

**Comment by KATT** [comment, marked as answer]
@kimroen this is tracked in #506. I haven't encountered this myself yet so haven't really cared to fix it, feel free to add a PR or put your ideas in there.
